src/validation.cpp

```diff
@@ -2085,23 +2086,34 @@ bool static FlushStateToDisk(const CChainParams& chainparams, CValidationState &
     static int64_t nLastFlush = 0;
     std::set<int> setFilesToPrune;
     bool full_flush_completed = false;
+
+    // we delay pruning when one of the blockfilterindex is syncing
+    bool delay_pruning = false;
+    ForEachBlockFilterIndex([&delay_pruning](BlockFilterIndex& index) {
+        if (!index.IsSynced()) { delay_pruning = true; }
```

Comment:
bf744288bd58718844841c088c498cd57a8f87d8 

nit, no need for `{}`. Alternative:
```
delay_pruning |= !index.IsSynced();
```

Reply:
Rather not do `|= !` for code readability, but removed the extra unnecessary block